### DISCUSSION TITLE
attaching character to moving platform with SetParent & block to slow down.

### DISCUSSION THREAD

**Discussion by ckdlskgkdrhd** [naverz/zepeto-studio-global#54] [Q&A]
howdy, I wrote a script to setparent on the block's trigger few days ago,
 which is a lot simpler to my likings compare to the provided moving block sample. (ughhh that world was just overly complicated)

my script was working fine on 1 dimensional movement.
when I used this script to perform 2 dimensional movement, x + y combined, it was squeezing? my character flat like a pancake. 
![캡처s](https://user-images.githubusercontent.com/110361648/185046166-d9d4afe0-0345-4166-9b8d-896a2e189b33.PNG)
![캡처z](https://user-images.githubusercontent.com/110361648/185046172-161eee58-ab5a-4898-a114-a544df8d7528.PNG)

also I kinda wanna make the moving block to slow down near the start and end points, to no success (totally expected)

how do I go about fixing them both.

---------------------------------movingcharacterwiththeblock--------------------------------------
`

    private zepetoCharacter : ZepetoCharacter; 
    Start(){
        ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
            this.zepetoCharacter = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character;
        });   
    }
    
    private OnTriggerEnter(col : Collider){
        if(this.zepetoCharacter == null || col.gameObject != this.zepetoCharacter.gameObject){
            return;       
        }
        this.zepetoCharacter.transform.SetParent(this.transform);
    }
    
    private OnTriggerExit(col : Collider){
        this.zepetoCharacter.transform.SetParent(null);
    }

**Comment by naverz-mkao** [comment]
Hi,

For your issue with your character getting squished, check to make sure you platform is scaled to (1,1,1). To make platforms that are not perfectly 1x1x1, you need to scale your collider, not the object itself. You can then have a child that is just a mesh and scale that to fit the collider.

You will basically get weird scaling issues when parenting and unparenting to a game object where the root transform is not set to 1x1x1.

With that being said, we recommend you use the moving block example. This is because the moving block example is taking into account physics, you may still get weird results by just parenting your player character to the moving platform.

Looks like you are having fun experimenting with the Zepeto platform, which we highly encourage!

**Comment by ckdlskgkdrhd** [comment]
haha thanks as always.
